runtime/lua/provider/perl/health.lua

```diff
@@ -0,0 +1,94 @@
+local health = vim.health
+
+local M = {}
+
+function M.check()
+  health.start('Perl provider (optional)')
+
+  if health.provider_disabled('perl') then
+    return
+  end
+
+  local perl_detect_table = vim.fn['provider#perl#Detect']()
+  local perl_exec = perl_detect_table[1]
+  local perl_warnings = perl_detect_table[2]
+
+  if perl_exec:find('^%s*$') then
+    if perl_warnings:find('%S') then
+      health.warn(perl_warnings, {
+        'See :help provider-perl for more information.',
+        'You may disable this provider (and warning) by adding `let g:loaded_perl_provider = 0` to your init.vim',
+      })
+    else
+      health.warn('No usable perl executable found')
+    end
+    return
+  end
+
+  health.info('perl executable: ' .. perl_exec)
+
+  -- we cannot use cpanm that is on the path, as it may not be for the perl
+  -- set with g:perl_host_prog
+  local ok = health.cmd_ok({ perl_exec, '-W', '-MApp::cpanminus', '-e', '' })
+  if not ok then
+    return { perl_exec, '"App::cpanminus" module is not installed' }
+  end
+
+  local latest_cpan_cmd = {
+    perl_exec,
+    '-MApp::cpanminus::fatscript',
+    '-e',
+    'my $app = App::cpanminus::script->new; $app->parse_options ("--info", "-q", "Neovim::Ext"); exit $app->doit',
+  }
+
+  local ok, latest_cpan = health.cmd_ok(latest_cpan_cmd)
+  if not ok or latest_cpan:find('^%s*$') then
+    health.error(
+      'Failed to run: ' .. table.concat(latest_cpan_cmd, ' '),
+      { "Make sure you're connected to the internet.", 'Are you behind a firewall or proxy?' }
+    )
+    return
+  elseif latest_cpan[1] == '!' then
+    local cpanm_errs = vim.split(latest_cpan, '!')
+    if cpanm_errs[1]:find("Can't write to ") then
+      local advice = {}
+      for i = 2, #cpanm_errs do
+        advice[#advice + 1] = cpanm_errs[i]
+      end
```

Comment:
I think this is trying to replicate `cpanm_errors[1:-2]`, but this doesn't look like it skips the last item for `advice` like the slice does